### ISSUE TITLE
Enhance interfaces to pbench-register-tool[-set]

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#1285]
Today `pbench-register-tool` and `pbench-register-tool-set` accept a single remote host in the `--remote` parameter.

Can we consider the following:

 * Change both `pbench-register-tool[-set]` so that the `--remote[s]` (adding the plural name) accepts a list of hosts separated by commas, or a file name containing a list of hosts
 * The list of remotes in a file can also have `hostname` / `label` pairs
 * Add a new option, `--no-install`, to avoid the remote install phase

Why would we want to do this?

Let's take a scenario where you have 7 tools you want registered on 10 hosts.  One can define that list of tools as a tool-set definition in their `pbench-agent.cfg` file named, `mytoolset`, and then perform the following:

```
$ for host in $(cat hosts.lis); do
      pbench-register-tool-set --toolset=mytoolset --remote=${host}
  done
```

This seems okay, except that `pbench-register-tool-set` will in turn invoke `pbench-register-tool` for each tool in the set, resulting in a sequence of 70 individual tool registrations via separate calls to `pbench-register-tool`.

If you then add the ability to label nodes, the above loop has to be broken down to loop by all the hosts with a given label, making it a bit more complicated.

If instead we could list the hosts in a file, and then just specify the lists of hosts once, e.g.:

```
pbench-register-tool-set --toolset=mytoolset --remotes=@hosts.lis
```

Then the user only has to define their tool set once in their `pbench-agent.cfg` file, create a file with a list of their hosts with the labels they want for each host, and invoke the `pbench-register-tool-set` once.

In turn, `pbench-register-tool` could then be invoked once per tool.  In our above scenario of 7 tools in our tool set with 10 hosts, `pbench-register-tool` would only be invoked 7 times.  Further, `pbench-register-tool` would be able to invoke all the remote registrations in parallel, if it so chooses (optional optimization).

Finally, by adding the `--no-install` parameter, the remote operation for registering a tool can be sped up to just be recording the tool information on the remote host.

**Comment by atheurer** [comment]
> Finally, by adding the `--no-install` parameter, the remote operation for registering a tool can be sped up to just be recording the tool information on the remote host.

When using pbench-register-tool[-set] --remote=<>, doesn't the remote operation still require the tool to be installed on the remote hosts?  Or may be I am forgetting a situation where that is not necessary?

**Comment by portante** [comment]
> When using pbench-register-tool[-set] --remote=<>, doesn't the remote operation still require the tool to be installed on the remote hosts?

When we'd invoke the remote `pbench-register-tool` operation remotely we'd also be passing along the `--no-install` option if it was given.

**Comment by portante** [comment]
Addressed by #1317.
